src/ucs/sys/sock.h

```diff
@@ -30,10 +30,24 @@ BEGIN_C_DECLS
 #define UCS_SOCKET_INET6_PORT(_addr) (((struct sockaddr_in6*)(_addr))->sin6_port)
 
 
-/* Returns an error if the default error handling should be
- * done (the error value will be returned to a caller),
- * otherwise `UCS_OK` */
-typedef ucs_status_t (*ucs_socket_io_err_cb_t)(void *arg, int io_errno);
+/**
+ * Error callback to handle errno and status of a given socket IO operation.
+ *
+ * @param [in] arg       User's argument for the error callback.
+ * @param [in] io_errno  Errno set for a given IO operation.
+ * @param [in] io_status Status set for a given IO operation.
+ *
+ * @return UCS_OK (UCS_ERR_CANCELED will be returned as a result of the IO
+ *         operation) if error handling was done in the callback and no other
+ *         actions are required from a caller, UCS_ERR_NO_PROGRESS if error
+ *         handling was done in the callback and the IO operation should be
+ *         continued (UCS_ERR_NO_PROGRESS will be retuned as the result
+ *         of the IO operation), otherwise - the default error handling
+ *         should be done and the returned status will be the result of
+ *         the IO operation.
+ */
+typedef ucs_status_t (*ucs_socket_io_err_cb_t)(void *arg, int io_errno,
+                                               ucs_status_t io_status);
```

Comment:
why do we need both errno and status?

Reply:
it can be useful if a user knows more information about what the error has to be returned from an IO operation; otherwise - just returns the same status

Reply:
isn't io_status calculated from io_errno anyway? 
pls pass only one (probably io_status)

Reply:
yes, `io_status` would be enough
done